src/OpenIddict.Client/OpenIddictClientConfiguration.cs

```diff
@@ -23,14 +24,27 @@ namespace OpenIddict.Client;
 public sealed class OpenIddictClientConfiguration : IPostConfigureOptions<OpenIddictClientOptions>
 {
     private readonly OpenIddictClientService _service;
+    private readonly IServiceProvider? _serviceProvider;
```

Comment:
Let's make this one non-nullable and update the existing constructor to throw (it's really meant for DI, no nobody should be using that 😄)